src/pybind/mgr/dashboard/frontend/src/app/core/auth/login/login.component.ts

```diff
@@ -48,16 +48,26 @@ export class LoginComponent implements OnInit {
             window.location.replace(login.login_url);
           }
         } else {
-          this.authStorageService.set(login.username, token, login.permissions, login.sso);
+          this.authStorageService.set(
+            login.username,
+            token,
+            login.permissions,
+            login.sso,
+            login.force_change_pwd
+          );
           this.router.navigate(['']);
         }
       });
     }
   }
 
   login() {
-    this.authService.login(this.model).then(() => {
-      this.router.navigate(['']);
+    this.authService.login(this.model).then((forceChangePwd: Boolean) => {
+      if (forceChangePwd) {
+        this.router.navigate(['/user-profile/edit']);
```

Comment:
Wouldn't it make sense to inform the user that he is forced to set a new password after first login? I'm wondering if it would confuse the user when he is redirected to the 'Change Password' page without notification. Maybe we could simply add a flag to the route, e.g.
```
this.router.navigate(['/user-profile/edit?force']);
```
to submit this state to the page and display a notification or banner there.

Reply:
Yes, I can add it. I think it will be more clear to the user.

Reply:
It looks as follow:
![image](https://user-images.githubusercontent.com/46715017/64159947-5b336d00-ce3b-11e9-9bb3-b3d4c73ca15c.png)